src/olympia/versions/tasks.py

```diff
@@ -112,11 +113,95 @@ def delete_preview_files(pk, **kw):
 
 @task
 @use_primary_db
-def extract_version_to_git(version_id, author_id=None, note=None):
-    """Extract a `File` into our git storage backend."""
+def extract_addon_to_git(addon_pk):
+    addon = Addon.unfiltered.get(pk=addon_pk)
+    log.info(
+        'Starting extraction of addon "{}" to git storage.'.format(addon_pk)
+    )
+
+    if addon.git_extraction_is_in_progress:
+        log.info('Aborting extraction of addon "{}" to git storage because it '
+                 'is already in progress.'.format(addon_pk))
+        return
+
+    # "Lock" this add-on so that the git extraction can only run once and tasks
+    # for extracting related versions are delayed until this task has finished.
+    git_storage, created = GitExtraction.objects.update_or_create(
+        addon=addon,
+        defaults={'in_progress': True},
+    )
+
+    # Filter out versions that are already present in the git storage.
+    versions = addon.versions.filter(git_hash='').order_by('created')
```

Comment:
You need to use `addon.versions(manager='unfiltered_for_relations')` to also include deleted versions.

Reply:
Also, if all we need are version pks, might as well use `values_list('pk', flat=True)` to avoid fetching objects needlessly. Especially as we're looping over all existing versions for this addon.

Reply:
> You need to use `addon.versions(manager='unfiltered_for_relations')` to also include deleted versions.

Interesting. This line comes from the "migrate to git" command, so does that mean we didn't git-extract the deleted versions?

Reply:
> Interesting. This line comes from the "migrate to git" command, so does that mean we didn't git-extract the deleted versions?

Yeah, and I don't think that's intended.